### PR TITLE
Fix relative files not mapped in error output

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This error seems to only occur when `singleBundle: false` is set in the config. When that happens the location in stack traces are often relative and not always absolute like they are with `singleBundle: true`. Our source map resolver didn't take that into account and would parse paths incorrectly (`foo/bar.js` -> `/bar.js` instead of `/foo/bar.js`). 

To resolve that the regex needed to be updated to be able to deal with relative paths and we need to construct the absolute path to get the correct object out of the bundler map.

### CODE REVIEW

File: test/fixtures/sourcemap-base-bundle/karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		process.cwd(),
+		"test",
+		"fixtures",
+		"sourcemap-base-bundle",
+		"files",
+		"main-a.js",
+	),
+].sort();
+
+const envPlugin = {
```
Comment (jridgewell) [inline test/fixtures/sourcemap-base-bundle/karma.conf.js:16]: I think all of this is unnecesary?

File: src/format-error.ts [hunk 2]
```diff
@@ -9,14 +9,15 @@ type Formatter = (m: string) => string;
...
+	basePath: string,
 	formatError?: Formatter,
 ) {
 	const consumers = new WeakMap<RawSourceMap, SourceMapConsumer>();
-	const regex = /((?:\b[A-Z]:)?\/[^ #?:]+)[^ :]*:(\d+):(\d+)/gi;
-	//             |            |          ||    ||    |^^^^^^ Column
-	//             |            |          ||    |^^^^^^ Line
-	//             |            |          |^^^^^^ Eat any reamining URL (query in particular)
-	//             ^^^^^^^^^^^^^^^^^^^^^^^^^ URL pathname extraction
+	const regex = /((?:\b[A-Z]:|[^ #?:(]+)?\/[^ #?:]+)[^ :]*:(\d+):(\d+)/gi;
```
Comment (jridgewell) [inline src/format-error.ts:16]: I think this is just preventing the opening `(` from being captured as part of the URL? If so, that could just modify the existing regex:

```suggestion
	const regex = /((?:\b[A-Z]:)?[^ #?:(]+)[^ :]*:(\d+):(\d+)/gi;
```

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: test/fixtures/sourcemap-base-bundle/files/main-a.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import { fetchPolyfill } from "../../../fetch-polyfill.js";
+// "env" module is synthetically created by the karma.conf.js using an esbuild plugin.
+import expectedSources from "env";
+
+describe("simple", () => {
```
[the remark this answers is not in the document]
Comment (marvinhagemeister) [reply test/fixtures/sourcemap-base-bundle/files/main-a.js:1]: Great idea! I've added such a test 👍

**Comment by marvinhagemeister** [comment]
Looks like the CI container has network problems right now.

**Review by jridgewell — APPROVED** [review]

**Pull request merged**